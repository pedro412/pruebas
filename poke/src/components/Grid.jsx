const Grid = ({ pokemons }) => {
  return (
    <>
      <h1>Grid</h1>

      {pokemons.length < 1 && (
        <div className="alert">No Pokemon captured yet</div>
      )}

      <div className="grid">
        {pokemons.map(({ name, id, image, count }) => (
          <div key={id} className="card">
            <h2>{name}</h2>
            <p>ID: {id}</p>
            <p>Count: {count}</p>
            <img src={image} alt={name} />
          </div>
        ))}
      </div>
    </>
  );
};

export default Grid;
