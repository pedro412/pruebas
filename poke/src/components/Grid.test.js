import Grid from './Grid';
const enzyme = require('enzyme');

const pokemonsMock = [
  {
    id: 25,
    name: 'pikachu',
    image:
      'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/25.png',
    count: 1,
  },
  {
    id: 143,
    name: 'snorlax',
    image:
      'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/143.png',
    count: 1,
  },
  {
    id: 448,
    name: 'lucario',
    image:
      'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/448.png',
    count: 1,
  },
];

describe('<Grid />', () => {
  const wrapper = enzyme.shallow(<Grid pokemons={[]} />);

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain alert if there are no pokemons', () => {
    expect(wrapper.find('.alert')).toHaveLength(1);
  });

  it('should render 3 cards if there are 3 pokemons', () => {
    const wrapper = enzyme.shallow(<Grid pokemons={pokemonsMock} />);
    expect(wrapper.find('.card')).toHaveLength(3);
  });
});
