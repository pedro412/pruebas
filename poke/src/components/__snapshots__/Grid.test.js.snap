// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> should render correctly 1`] = `
<Fragment>
  <h1>
    Grid
  </h1>
  <div
    className="alert"
  >
    No Pokemon captured yet
  </div>
  <div
    className="grid"
  />
</Fragment>
`;
